src/hyp3_isce2/burst.py

```diff
@@ -303,30 +309,34 @@ def get_asf_session() -> requests.Session:
 
 
 def download_bursts(param_list: Iterator[BurstParams]) -> List[BurstMetadata]:
-    """Download bursts and create SAFE files.
+    """Download bursts in parallel and creates SAFE files.
 
     For each burst:
         1. Download metadata
-        2. Create BurstMetadata object
-        3. Create directory structure
-        4. Write metadata
-        5. Download and write geotiff
+        2. Download geotiff
+        3. Create BurstMetadata object
+        4. Create directory structure
+        5. Write metadata
+        6. Move geotiff to correct directory
 
     Args:
         param_list: An iterator of burst search parameters.
 
     Returns:
         A list of BurstMetadata objects.
     """
-    bursts = []
     with get_asf_session() as asf_session:
-        for i, params in enumerate(param_list):
-            print(f'Creating SAFE {i+1}...')
-            metadata_xml = download_metadata(asf_session, params)
-            burst = BurstMetadata(metadata_xml, params)
-            spoof_safe(asf_session, burst)
-            bursts.append(burst)
+        with ThreadPoolExecutor(max_workers=10) as executor:
+            xml_futures = [executor.submit(download_metadata, asf_session, params) for params in param_list]
+            tiff_futures = [executor.submit(download_burst, asf_session, params) for params in param_list]
+            metadata_xmls = [future.result() for future in xml_futures]
+            burst_paths = [future.result() for future in tiff_futures]
```

Comment:
Cool! Be nice to add this to the SDK too: https://github.com/ASFHyP3/hyp3-sdk/issues/113